### DISCUSSION TITLE
Questions about some api's compatibility

### DISCUSSION THREAD

**Discussion by Yahui-Li** [google/oboe#1434] [Q&A] [answered]
At the last of doc "Full Guide To Oboe" said that: 
getFramesRead(), getFramesWritten(), getTimestamp() will return Result::ErrorUnimplemented for OpenSLES streams, But I found that in AudioStream.cpp, there has a implementation of getFramesRead():

```
int64_t AudioStream::getFramesRead() {
    updateFramesRead();
    return mFramesRead;
}
```

and in  AudioOutputStreamOpenSLES.cpp, function updateFramesRead are also be implemented.
```
void AudioOutputStreamOpenSLES::updateFramesRead() {
    if (usingFIFO()) {
        AudioStreamBuffered::updateFramesRead();
    } else {
        mFramesRead = getFramesProcessedByServer();
    }
}
```

does it means those three functions(getFramesRead, getFramesWritten, getTimestamp) is also available in OpenSLES streams before OS version android O？

**Comment by dturner** [comment, marked as answer]
Thanks for spotting this. We'll update the docs. Issue filed: https://github.com/google/oboe/issues/1444
